### PR TITLE
Allow MTU to be specified on Route

### PR DESCRIPTION
**Pull request by mccv1r0**
Allow MTU to be specified on Route

Signed-off-by: Michael Cambria <mcambria@redhat.com>

### CODE REVIEW

**Review by bboreham — APPROVED** [review]

**Comment by dcbw** [comment]
@mccv1r0 can you add the new field to SPEC.md? Also we'll need to move this struct and associated functions into pkg/types/100/ with MTU, and then into 020 and 040 without the MTU bits.

**Review by warrengunn48 — APPROVED** [review]

File: pkg/types/100/types.go [hunk 1]
```diff
@@ -305,3 +305,54 @@ func (c *IPConfig) UnmarshalJSON(data []byte) error {
 	c.Gateway = ipc.Gateway
 	return nil
 }
+
+type Route struct {
+	Dst net.IPNet
+	GW  net.IP
+	MTU int
```
Comment (mars1024) [inline pkg/types/100/types.go:312]: Is it better to use int pointer here since MTU is optional?

File: pkg/types/100/types.go [hunk 2]
```diff
@@ -305,3 +305,54 @@ func (c *IPConfig) UnmarshalJSON(data []byte) error {
...
+		return err
+	}
+
+	r.Dst = net.IPNet(rt.Dst)
+	r.GW = rt.GW
+	r.MTU = rt.MTU
+	return nil
+}
+
+func (r Route) MarshalJSON() ([]byte, error) {
```
Comment (mars1024) [inline pkg/types/100/types.go:350]: ditto

File: pkg/types/040/types.go [hunk 3]
```diff
@@ -304,3 +304,49 @@ func (c *IPConfig) UnmarshalJSON(data []byte) error {
...
+	if err := json.Unmarshal(data, &rt); err != nil {
+		return err
+	}
+
+	r.Dst = net.IPNet(rt.Dst)
+	r.GW = rt.GW
+	return nil
+}
+
+func (r Route) MarshalJSON() ([]byte, error) {
```
Comment (mars1024) [inline pkg/types/040/types.go:345]: ditto

**Comment by dcbw** [comment]
@mccv1r0 need to remove `Route` from pkg/types/types.go and that'll give you a cascade of compile issues. Which basically means you need to change each version's Route to it's own rather than types.Route, eg for 020:

```
type IPConfig struct {
	IP      net.IPNet
	Gateway net.IP
-	Routes  []types.Route
+	Routes  []Route
}
```

**Comment by squeed** [comment]
closing this for now - @mccv1r0, feel free to re-open this if we should pick this back up.

**Pull request closed without merging**